### PR TITLE
Feature/us314 sign up family members

### PR DESCRIPTION
**Pull request by shankx50**
@IngageKrizJ @IngageRitterS This is for pre-code review and pre-testing. Don't merge yet.

### CODE REVIEW

File: Gateway/MinistryPlatform.Translation/Services/GroupService.cs [hunk 1]
```diff
@@ -29,14 +31,7 @@ public GroupService(IMinistryPlatformService ministryPlatformService)
...
         {
             logger.Debug("Adding participant " + participantId + " to group " + groupId);
-
-            // TODO Basing "Full" on Group_Is_Full flag, pending outcome of SPIKE: US1080
-            Group g = getGroupDetails(groupId);
-            if (g.Full)
-            {
-                throw (new GroupFullException(g));
-            }
-
```
Comment (IngageKrizJ) [inline Gateway/MinistryPlatform.Translation/Services/GroupService.cs]: @IngageRitterS, why was this check removed?  Don't we still want to check for full when adding a participant?

File: Gateway/common.config [hunk 2]
```diff
@@ -44,7 +48,7 @@
   <!--Below are the credentials of the user that registers new records via the register form.-->
   <!--TODO Remove credentials out into a gitignore file-->
   <add key="ApiUser" value="register_api"/>
-  <add key="ApiPass" value="01152015$j"/>
+  <add key="ApiPass" value="f*xe6awehu"/>
```
Comment (IngageKrizJ) [inline Gateway/common.config]: We don't want this new password to get into the development branch - I think Tony has removed the ApiUser and ApiPass keys completely from this file.

File: Gateway/crds-angular/Models/Crossroads/PartID.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace crds_angular.Models.Crossroads
+{
+    public class PartID
```
Comment (IngageKrizJ) [inline Gateway/crds-angular/Models/Crossroads/PartID.cs]: @IngageRitterS Should this be called "Participant", with a property of "ParticipantID"?

File: crossroads.net/app/community_groups_signup/group_signup_controller.js [hunk 4]
```diff
@@ -18,10 +18,30 @@ require('../services/group_service');
         vm.alreadySignedUp = false;
         vm.viewReady = false;
         vm.modalInstance = {};
+        vm.response = {};
+
+        vm.signup = function(){
+            var test = hasParticipantID(vm.response);
+            console.log(test);
```
Comment (IngageKrizJ) [inline crossroads.net/app/community_groups_signup/group_signup_controller.js]: @shankx50 Probably should remove this console.log before merging to development

File: Gateway/MinistryPlatform.Models/ContactRelationship.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+
+namespace MinistryPlatform.Models
+{
+    public class ContactRelationship
```
Comment (IngageMaddoxT) [inline Gateway/MinistryPlatform.Models/ContactRelationship.cs]: :+1:

File: Gateway/crds-angular/Controllers/API/GroupController.cs [hunk 6]
```diff
@@ -37,43 +49,53 @@ public class GroupController : MPAuth
          */
 
         [ResponseType(typeof (GroupDTO))]
-        [Route("api/group/{groupId}/user")]
-        public IHttpActionResult Post(String groupId)
+        [Route("api/group/{groupId}/participants")]
+        public IHttpActionResult Post(int groupId, [FromBody] PartID partId )
```
Comment (IngageMaddoxT) [inline Gateway/crds-angular/Controllers/API/GroupController.cs]: This is a lot of logic for a controller.

File: Gateway/MinistryPlatform.Translation/Services/GroupService.cs (hunk 1 above)
Comment (IngageRitterS) [inline Gateway/MinistryPlatform.Translation/Services/GroupService.cs]: @IngageKrizJ  - we were doing the check twice. In the GroupController, the first thing that is done on a Post is a call to getgroupdetails.  Then we add the participant(s) to the group if there is space available.

File: Gateway/crds-angular/Controllers/API/GroupController.cs (hunk 6 above)
Comment (IngageRitterS) [inline Gateway/crds-angular/Controllers/API/GroupController.cs]: @IngageMaddoxT I completely agree.  Suggestions??

File: crossroads.net/app/community_groups_signup/group_signup_controller.js [hunk 7]
```diff
@@ -18,10 +21,44 @@ require('../services/group_service');
...
         vm.viewReady = false;
         vm.modalInstance = {};
+        vm.response = {};
+        vm.formValid = true;
+
+        vm.signup = function(form){
+            var test = hasParticipantID(vm.response);
+            var flag = false;
+            for(var i = 0; i < vm.response.length;i++){
+                console.log(vm.response[i]['newAdd']);
```
Comment (IngageKrizJ) [inline crossroads.net/app/community_groups_signup/group_signup_controller.js]: Remove console.log

File: crossroads.net/app/community_groups_signup/group_signup_controller.js [hunk 8]
```diff
@@ -32,6 +69,8 @@ require('../services/group_service');
         }, function () {
             if (pageRequest.pages.length > 0) {
                 vm.signupPage = pageRequest.pages[0];
+                console.log("CMS");
```
Comment (IngageKrizJ) [inline crossroads.net/app/community_groups_signup/group_signup_controller.js]: Remove console.log

File: crossroads.net/app/community_groups_signup/group_signup_controller.js [hunk 9]
```diff
@@ -124,5 +150,41 @@ vm.signup = function(){
...
+                    }
+                }
+            }else{
+                if(array.SignUpFamilyMembers[0]['userInGroup'] === false){
+                        result = false;
+                    }else{
+                        result = true;
+                    }
+            }
+            console.log("allSignedUp: "+result);
```
Comment (IngageKrizJ) [inline crossroads.net/app/community_groups_signup/group_signup_controller.js]: Remove console.log

**Comment by IngageKrizJ** [comment]
We are going to merge this one, and we've created another story (US1212 - https://rally1.rallydev.com/#/27593764268d/detail/userstory/33224565014) to refactor the GroupController into service(s).

**Pull request merged**